packages/backend/src/open_payments/auth/middleware.test.ts

```diff
@@ -47,25 +63,74 @@ describe('Auth Middleware', (): void => {
       action: AccessAction.Read
     })
     const authOpenApi = await deps.use('authOpenApi')
+    requestPath = '/introspect'
     validateRequest = authOpenApi.createRequestValidator({
-      path: '/introspect',
+      path: requestPath,
       method: HttpMethod.POST
     })
     grantReferenceService = await deps.use('grantReferenceService')
+    const { publicKey, privateKey } = await generateTestKeys()
+    requestMethod = HttpMethod.POST.toUpperCase() as RequestMethod
+    requestBody = {
+      access_token: token,
+      proof: 'httpsig',
+      resource_server: 'test'
+    }
+    requestAuthorization = `GNAP ${token}`
+    requestUrl = Config.authServerGrantUrl + requestPath //'http://127.0.0.1:3006/introspect'//authServerIntrospectionUrl.toString()//'http://example.com/introspect'//authServerIntrospectionUrl.toString()
+    requestSignatureHeaders = await generateSigHeaders(
+      privateKey,
+      requestUrl,
+      requestMethod,
+      {
+        body: requestBody,
+        authorization: requestAuthorization
+      }
+    )
+    requestJwk = publicKey
   })
 
+  function setupHttpSigContext(options: SetupOptions): HttpSigContext {
+    const context = setup(options)
+    if (
+      !context.headers['signature'] ||
+      !context.request.headers['signature']
+    ) {
+      throw new Error('missing signature header')
+    }
+    if (
+      !context.headers['signature-input'] ||
+      !context.request.headers['signature-input']
+    ) {
+      throw new Error('missing signature-input header')
+    }
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return context as any
+  }
+
   beforeEach(async (): Promise<void> => {
-    ctx = setup({
+    ctx = setupHttpSigContext({
       reqOpts: {
         headers: {
           Accept: 'application/json',
-          Authorization: `GNAP ${token}`
-        }
+          Authorization: `GNAP ${token}`,
+          Signature: `sig1=:${requestSignatureHeaders.signature}:`,
+          'Signature-Input': requestSignatureHeaders.sigInput,
+          'Content-Digest': requestSignatureHeaders.contentDigest,
+          'Content-Length': JSON.stringify(requestBody).length.toString()
+        },
+        method: requestMethod,
+        body: requestBody,
```

Comment:
I know it'd require a bit of rearranging, but I think we should test requests with and without a body (and `Content-*` headers) (at least for the signature verification tests).